weaviate/collections/batch/base.py

```diff
@@ -284,69 +257,54 @@ def _shutdown(self) -> None:
-                        ):
-                            time.sleep(1)
-                            continue
 
-                    self.__time_stamp_last_request = time.time()
+            if (
+                self.__active_requests < self.__concurrent_requests
+                and len(self.__batch_objects) + len(self.__batch_references) > 0
+            ):
+                self.__time_stamp_last_request = time.time()
 
-                if self.__active_requests < self.__concurrent_requests and (
-                    len(self.__batch_objects) > 0 or len(self.__batch_references) > 0
-                ):
-                    self._batch_send = True
-                    self.__active_requests_lock.acquire()
-                    self.__active_requests += 1
-                    self.__active_requests_lock.release()
-
-                    objs = self.__batch_objects.pop_items(self.__recommended_num_objects)
-                    self.__uuid_lookup_lock.acquire()
-                    refs = self.__batch_references.pop_items(
-                        self.__recommended_num_refs, uuid_lookup=self.__uuid_lookup
-                    )
-                    self.__uuid_lookup_lock.release()
-                    # do not block the thread - the results are written to a central (locked) list and we want to have multiple concurrent batch-requests
-                    asyncio.run_coroutine_threadsafe(
-                        self.__send_batch_async(
-                            objs,
-                            refs,
-                            readd_rate_limit=isinstance(self.__batching_mode, _RateLimitedBatching),
-                        ),
-                        loop,
-                    )
+                self._batch_send = True
+                self.__active_requests_lock.acquire()
+                self.__active_requests += 1
+                self.__active_requests_lock.release()
 
-                time.sleep(refresh_time)
+                objs = self.__batch_objects.pop_items(self.__recommended_num_objects)
+                self.__uuid_lookup_lock.acquire()
+                refs = self.__batch_references.pop_items(
+                    self.__recommended_num_refs, uuid_lookup=self.__uuid_lookup
+                )
+                self.__uuid_lookup_lock.release()
+                # do not block the thread - the results are written to a central (locked) list and we want to have multiple concurrent batch-requests
+                self.__loop.schedule(
+                    self.__send_batch,
+                    objs,
+                    refs,
+                    readd_rate_limit=isinstance(self.__batching_mode, _RateLimitedBatching),
+                )
 
-        finally:
-            future = asyncio.run_coroutine_threadsafe(self.__connection.aclose(), loop)
-            future.result()  # Wait for self._connection.aclose() to finish
-            loop.call_soon_threadsafe(loop.stop)
+            time.sleep(refresh_time)
```

Comment:
The event loop is now managed at the client level instead of within each batch. When a batching job is created, it just fetches the current event loop without spawning its own. Therefore, connection opening and closing is handled at the client-level in the `client.connect()` method and so no longer occurs in the batch init processes

Reply:
ah sorry - you mentioned there was a bug in the batch loop

Reply:
Ah yeh, also that!

Reply:
What was the bug? :)

Reply:
There was a timing issue in the rate limiting workflow due to some of the logic. Due to the time taken to spin up the event loop, it wasn't apparent. However, with the event loop managed elsewhere, this delay doesn't occur and the error surfaces. I'll check again if it's still present

Reply:
great, let's resolve this after you confirmend it is gone!

Reply:
Have added tests for rate limiting on large and small ingestion sizes and there are no issues :D